File: src/test/java/com/ject/studytrip/mission/helper/DailyMissionTestHelper.java
package com.ject.studytrip.mission.helper;

import com.ject.studytrip.mission.domain.model.DailyMission;
import com.ject.studytrip.mission.domain.model.Mission;
import com.ject.studytrip.mission.domain.repository.DailyMissionRepository;
import com.ject.studytrip.mission.fixture.DailyMissionFixture;
import com.ject.studytrip.trip.domain.model.DailyGoal;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class DailyMissionTestHelper {

    @Autowired private DailyMissionRepository dailyMissionRepository;

    public DailyMission saveDailyMission(Mission mission, DailyGoal dailyGoal) {
        DailyMission dailyMission = DailyMissionFixture.createDailyMission(mission, dailyGoal);
        return dailyMissionRepository.save(dailyMission);
    }

    public DailyMission saveDeletedDailyMission(Mission mission, DailyGoal dailyGoal) {
        DailyMission dailyMission = DailyMissionFixture.createDailyMission(mission, dailyGoal);
        dailyMission.updateDeletedAt();

        return dailyMissionRepository.save(dailyMission);
    }
}

File: src/test/kotlin/com/ject/studytrip/mission/helper/DailyMissionTestHelper.kt
package com.ject.studytrip.mission.helper

import com.ject.studytrip.mission.domain.model.DailyMission
import com.ject.studytrip.mission.domain.model.Mission
import com.ject.studytrip.mission.domain.repository.DailyMissionRepository
import com.ject.studytrip.mission.fixture.DailyMissionFixture
import com.ject.studytrip.trip.domain.model.DailyGoal
import org.springframework.stereotype.Component

@Component
class DailyMissionTestHelper(
    private val dailyMissionRepository: DailyMissionRepository,
) {
    fun saveDailyMission(
        mission: Mission,
        dailyGoal: DailyGoal,
    ): DailyMission = dailyMissionRepository.save(DailyMissionFixture(mission, dailyGoal).create())

    fun saveDeletedDailyMission(
        mission: Mission,
        dailyGoal: DailyGoal,
    ): DailyMission = dailyMissionRepository.save(DailyMissionFixture(mission, dailyGoal).create().also { it.updateDeletedAt() })
}
